Mathlib/AlgebraicGeometry/Restrict.lean

```diff
@@ -34,55 +34,123 @@ variable {C : Type u₁} [Category.{v} C]
+@[simp]
+lemma range_ι : Set.range U.ι.val.base = U :=
+  Subtype.range_val
+
+lemma ι_image_top : U.ι ''ᵁ ⊤ = U :=
   U.openEmbedding_obj_top
 
-instance Scheme.ιOpens_appLE_isIso {X : Scheme.{u}} (U : Opens X) :
-    IsIso ((Scheme.ιOpens U).appLE U ⊤ (by simp)) := by
-  simp only [restrict_presheaf_obj, ofRestrict_appLE]
-  have : IsIso (homOfLE <| le_of_eq (ιOpens_image_top U)) :=
-    homOfLE_isIso_of_eq _ <| ιOpens_image_top U
+@[simp]
+lemma ι_preimage_self : U.ι ⁻¹ᵁ U = ⊤ :=
+  Opens.inclusion_map_eq_top _
+
+instance ι_appLE_isIso :
+    IsIso (U.ι.appLE U ⊤ U.ι_preimage_self.ge) := by
+  simp only [ι, ofRestrict_appLE]
+  show IsIso (X.presheaf.map (eqToIso U.ι_image_top).hom.op)
   infer_instance
 
-lemma Scheme.ofRestrict_app_self {X : Scheme.{u}} (U : Opens X) :
-    (ιOpens U).app U = X.presheaf.map (eqToHom (by simp)).op := rfl
+lemma ι_app_self : U.ι.app U = X.presheaf.map (eqToHom (X := U.ι ''ᵁ _) (by simp)).op := rfl
 
-lemma Scheme.eq_restrict_presheaf_map_eqToHom {X : Scheme.{u}} (U : Opens X) {V W : Opens U}
-    (e : Scheme.ιOpens U ''ᵁ V = ιOpens U ''ᵁ W) :
-  X.presheaf.map (eqToHom e).op =
-    (X ∣_ᵤ U).presheaf.map (eqToHom <| U.openEmbedding.functor_obj_injective e).op := rfl
+lemma eq_presheaf_map_eqToHom {V W : Opens U} (e : U.ι ''ᵁ V = U.ι ''ᵁ W) :
+    X.presheaf.map (eqToHom e).op =
+      U.toScheme.presheaf.map (eqToHom <| U.openEmbedding.functor_obj_injective e).op := rfl
 
 @[simp]
-lemma opensRange_ιOpens {X : Scheme.{u}} (U : Opens X) : (Scheme.ιOpens U).opensRange = U :=
-  Opens.ext Subtype.range_val
+lemma nonempty_iff : Nonempty U.toScheme ↔ (U : Set X).Nonempty := by
+  simp only [toScheme_carrier, SetLike.coe_sort_coe, nonempty_subtype]
+  rfl
 
 attribute [-simp] eqToHom_op in
 /-- The global sections of the restriction is isomorphic to the sections on the open set. -/
 @[simps!]
-def Scheme.resTop (X : Scheme.{u}) (U : Opens X) : Γ(X ∣_ᵤ U, ⊤) ≅ Γ(X, U) :=
-  X.presheaf.mapIso (eqToIso U.openEmbedding_obj_top.symm).op
+def topIso : Γ(U, ⊤) ≅ Γ(X, U) :=
+  X.presheaf.mapIso (eqToIso U.ι_image_top.symm).op
+
+/-- The stalks of an open subscheme is isomorphic to the stalk of the original scheme. -/
+def stalkIso {X : Scheme.{u}} (U : X.Opens) (x : U) :
+    U.toScheme.presheaf.stalk x ≅ X.presheaf.stalk x.1 :=
+  X.restrictStalkIso (Opens.openEmbedding _) _
+
+@[reassoc (attr := simp)]
+lemma germ_stalkIso_hom {X : Scheme.{u}} (U : X.Opens)
+    {V : TopologicalSpace.Opens U} (x : V) :
+      U.toScheme.presheaf.germ x ≫ (U.stalkIso x.1).hom =
+        X.presheaf.germ ⟨x.1.1, show x.1.1 ∈ U.ι ''ᵁ V from ⟨x.1, x.2, rfl⟩⟩ :=
+    PresheafedSpace.restrictStalkIso_hom_eq_germ _ _ _ _ _
+
+end Scheme.Opens
+
+/-- If `U` is a family of open sets that covers `X`, then `X.restrict U` forms an `X.open_cover`. -/
+@[simps! J obj map]
+def Scheme.openCoverOfSuprEqTop {s : Type*} (X : Scheme.{u}) (U : s → X.Opens)
+    (hU : ⨆ i, U i = ⊤) : X.OpenCover where
```

Comment:
```suggestion
def Scheme.openCoverOfiSupEqTop {s : Type*} (X : Scheme.{u}) (U : s → X.Opens)
    (hU : ⨆ i, U i = ⊤) : X.OpenCover where
```

Reply:
This is copy pasted from elsewhere. I'll make this change in a follow-up PR.